content/helpers.go

```diff
@@ -230,8 +230,31 @@ func seekReader(r io.Reader, offset, size int64) (io.Reader, error) {
 }
 
 func copyWithBuffer(dst io.Writer, src io.Reader) (written int64, err error) {
-	buf := bufPool.Get().(*[]byte)
-	written, err = io.CopyBuffer(dst, src, *buf)
-	bufPool.Put(buf)
+	bufRef := bufPool.Get().(*[]byte)
```

Comment:
Should note that `io.Copy` does some other things here, specifically checks for `WriterTo` and `ReadFrom` interfaces, which themselves have their own optimizations, including in some cases doing all copying in kernel.